### ISSUE TITLE
Braces/brackets/parentheses matching is slow on big files

### ISSUE THREAD

**Issue by andrey-dresvyannikov** [Microsoft/vscode#19794]
- VSCode Version: 1.9.0
- OS Version: xubuntu 16.10

About:
```
Version 1.9.0
Commit 27240e71ef390bf2d66307e677c2a333cebf75af
Date 2017-02-02T08:21:13.228Z
Shell 1.4.6
Renderer 53.0.2785.143
Node 6.5.0
```
Steps to Reproduce:

1. Open big file (i have ~2000 lines)
2. Go to the center/end of file
3. Try typing closing brace/bracket/parenthesis and type anything after

the stuff you typed after closing brace/bracket/parenthesis will be inserted after several seconds

**Comment by alexandrudima** [comment]
@andrey-dresvyannikov What programming language are you using? Are you able to share the file that reproduces the problem or a similar file?

**Comment by andrey-dresvyannikov** [comment]
I'm using python 3.6. Unfortunately i'm not allowed to share the file (due to NDA).
Though I can try to find opensource project with huge file and try to reproduce it there, if you wish.

**Comment by alexandrudima** [comment]
I can reproduce some slowness (70ms), but not at the same magnitude you experience (several seconds). That's why I'd be inderested in the file characteristics (e.g. size on disk).

Closing an unbalanced closing bracket will lead to the entire text above being searched for the opening one (only to never be found).

Out of the box we define {}, () and [] as brackets for python. Did you perhaps install an extension that would define additional bracket pairs? (The bracket pairs end up in constructed regex which might become more expensive depending on the bracket pairs).

I've tried reproducing by repeating the following snippet until I got 4k lines and a size on disk of 170KB.

```py
while babies < 100:
    print 'This generation has {0} babies'.format(babies)
    parents, babies = (babies, parents + babies)
```

TL;DR: Do you have an extension that might impact this? Can you point me to a file size/shape that brings us in the same ballpark of time spent in bracket matching.

**Comment by andrey-dresvyannikov** [comment]
@alexandrudima (sad to say that but) file is 2273 lines and is ~100KB (and it really takes seconds to display typed text).

I have following extensions installed: 
* elm 0.8.3 
* Emblem.js Syntax 1.0.0
* LiveScript Language Support 0.1.1
* Python 0.5.8 
* Sass 1.3.0 
* Vim 0.5.2

PS:  You can provide me with steps to measure time/get proper output (if it possible), so I can attach valuable info.

**Comment by alexandrudima** [comment]
@andrey-dresvyannikov 

Given your extension list, does the issue still occur when you launch VS Code w/ extensions disabled? From the command line: `code --disable-extensions`

**Comment by andrey-dresvyannikov** [comment]
@alexandrudima you're right - the problems is with extensions. With extensions turned off - it works like charm.

**Comment by alexandrudima** [comment]
@andrey-dresvyannikov The next steps here would be to enable/disable extensions and find out which one is causing the slowness and follow up with an issue against that extension. 

I'm sorry at this time it is a manual process, as we don't have an easy way to profile the extension host and figure out immediately in which extension the time is spent.

**Comment by andrey-dresvyannikov** [comment]
@alexandrudima yep, i see. Thank you for your time.
